app/controllers/application_controller.rb

```diff
@@ -1,2 +1,36 @@
 class ApplicationController < ActionController::Base
+  # Is the user signed in?
+  # @return [Boolean]
+  def user_signed_in?
+    session[:userinfo].present?
+  end
+
+  # Set the @current_user or redirect to public page
+  def authenticate_user!
+    # Redirect to page that has the login here
+    if user_signed_in?
+      @current_user = session[:userinfo]
+    else
+      session[:redirect_path] = request&.fullpath
+      redirect_to login_url
+    end
+  end
+
+  # What's the current_user?
+  # @return [Hash]
+  def current_user
+    @current_user
+  end
+
+  # @return the path to the login page
+  def login_url
+    '/auth/cognito-idp/'
+  end
+
+  def logout_url
```

Comment:
Actions like this should be moved in application_helper.rb

Also you can use rails' [url helpers](https://api.rubyonrails.org/v3.2/classes/ActionView/Helpers/UrlHelper.html) for building up your urls.